core/jvm/src/main/scala-3/zio/stm/ZSTMLockSupport.scala

```diff
@@ -0,0 +1,88 @@
+package zio.stm
+
+import zio.stm.ZSTM.internal.{LockTimeoutMaxMicros, LockTimeoutMinMicros}
+
+import java.util.concurrent.locks.ReentrantLock
+import java.util.concurrent.{ThreadLocalRandom, TimeUnit}
+
+private object ZSTMLockSupport {
+
+  type Lock = ReentrantLock
+  object Lock {
+    def apply(fair: Boolean = false): Lock = new ReentrantLock(fair)
+  }
+
+  inline def lock[A](refs: collection.Set[TRef[?]])(inline f: A)(implicit rnd: ThreadLocalRandom): Boolean =
```

Comment:
As a general question, should we use `using/given` in Scala 3 specific files?

Reply:
I thought of it, the only issue with it is that when we use `using`, we can't pass the implicit like this raises a compiler error: 

```scala
lock(???)(???)(rnd)
```

It has to be changed to this to work which we can't cross-compile

```scala 3
lock(???)(???)(using rnd)
```

But since here we're not passing the implicit anywhere I think we can change it